gax-grpc/src/main/java/com/google/api/gax/grpc/OperationFuture.java

```diff
@@ -27,116 +27,28 @@
-    implements ApiFuture<ResponseT> {
-  private final Object lock = new Object();
-
-  private final RetryingFuture<Operation> pollingFuture;
-  private final ApiFuture<Operation> initialFuture;
-  private final ApiFuture<ResponseT> resultFuture;
-  private final ApiFunction<Operation, MetadataT> metadataTransformer;
-
-  private volatile ApiFuture<Operation> peekedAttemptResult;
-  private volatile ApiFuture<MetadataT> peekedPollResult;
-  private volatile ApiFuture<Operation> gottenAttemptResult;
-  private volatile ApiFuture<MetadataT> gottenPollResult;
-
-  /**
-   * Creates a new operation future instance.
-   *
-   * @param pollingFuture retrying future which tracks polling of the server operation (in most
-   *     cases with exponential upper bounded intervals)
-   * @param initialFuture the initial future which started the operation on the server side
-   * @param responseTransformer transformer which unpacks a {@link ResponseT} object from an {@link
-   *     Operation} object
-   * @param metadataTransformer transformer which unpacks a {@link MetadataT} object from an {@link
-   *     Operation} object
-   */
-  public OperationFuture(
-      RetryingFuture<Operation> pollingFuture,
-      ApiFuture<Operation> initialFuture,
-      ApiFunction<Operation, ResponseT> responseTransformer,
-      ApiFunction<Operation, MetadataT> metadataTransformer) {
-    this.pollingFuture = checkNotNull(pollingFuture);
-    this.initialFuture = checkNotNull(initialFuture);
-    this.resultFuture = ApiFutures.transform(pollingFuture, responseTransformer);
-    this.metadataTransformer = checkNotNull(metadataTransformer);
-  }
-
-  @Override
-  public void addListener(Runnable listener, Executor executor) {
-    pollingFuture.addListener(listener, executor);
-  }
-
-  @Override
-  public boolean cancel(boolean mayInterruptIfRunning) {
-    return pollingFuture.cancel(mayInterruptIfRunning);
-  }
-
-  @Override
-  public boolean isCancelled() {
-    return pollingFuture.isCancelled();
-  }
-
-  @Override
-  public boolean isDone() {
-    return pollingFuture.isDone();
-  }
-
-  @Override
-  public ResponseT get() throws InterruptedException, ExecutionException {
-    pollingFuture.get();
-    return resultFuture.get();
-  }
-
-  @Override
-  public ResponseT get(long timeout, TimeUnit unit)
-      throws InterruptedException, ExecutionException, TimeoutException {
-    pollingFuture.get(timeout, unit);
-    return resultFuture.get();
-  }
-
+public interface OperationFuture<ResponseT, MetadataT, OperationT> extends ApiFuture<ResponseT> {
   /**
-   * Returns the value of {@code Operation.name} from the initial Operation object returned from the
-   * initial call to start the {@code Operation}. Blocks if the initial call to start the {@code
-   * Operation} hasn't returned yet.
+   * Returns the value of Operation.name from the initial Operation object returned from the initial
```

Comment:
Why did you remove {@code } blocks? Probably this should be rewritten in pure english (Operation.name -> operation name), since Operation now a type parameter with undefined properties.